src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -2904,116 +2890,58 @@ private static bool IsReallyAType(TypeSymbol type)
                 (type.SpecialType != SpecialType.System_Void);
         }
 
-        private struct Candidates
+        private static void GetAllCandidates(Dictionary<TypeSymbolWithAnnotations, TypeSymbolWithAnnotations> candidates, ArrayBuilder<TypeSymbolWithAnnotations> builder)
```

Comment:
The `Candidates` struct was added in this feature branch. The changes here include reverting to the approach used in `master` of several `static` methods that take a `Dictionary` of candidate types.